spec/checkPullRequestTemplateSpec.js

```diff
@@ -0,0 +1,471 @@
+      expect(github.issues.createComment).toHaveBeenCalled();
+      expect(github.issues.createComment).toHaveBeenCalledWith({
+        issue_number: payloadData.payload.pull_request.number,
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        body:
+          'Hi @' +
+          payloadData.payload.pull_request.user.login +
+          ', the body of this PR is missing the required description, ' +
+          'please update the body with a description of what this PR does.<br>' +
+          'Also, the body of this PR is missing the checklist section, please ' +
+          'update it to include the checklist. Thanks!',
+      });
+    });
+
+    it('should assign PR author', () => {
+      expect(github.issues.addAssignees).toHaveBeenCalled();
+      expect(github.issues.addAssignees).toHaveBeenCalledWith({
+        issue_number: payloadData.payload.pull_request.number,
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        assignees: [payloadData.payload.pull_request.user.login],
+      });
+    });
+  });
+
+  describe('when pull request contains description without filled checklist', () => {
+    beforeEach(async () => {
+      payloadData.payload.pull_request.body = bodyWithExplanation;
+      payloadData.payload.pull_request.maintainer_can_modify = false;
+      await robot.receive(payloadData);
+    });
+
+    it('should check the template', () => {
+      expect(checkPullRequestTemplateModule.checkTemplate).toHaveBeenCalled();
+    });
+
+    it('should ping PR author', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+      expect(github.issues.createComment).toHaveBeenCalledWith({
+        issue_number: payloadData.payload.pull_request.number,
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        body:
+          'Hi @' +
+          payloadData.payload.pull_request.user.login +
+          ', the karma and linter checklist has not been checked, ' +
+          'please make sure to run the frontend tests and lint tests before ' +
+          'pushing. The allow edits from maintainers checklist needs to ' +
+          'be ticked so that maintainers can rerun failed tests. Endeavour ' +
+          'to add this by ticking on the check box. Thanks!',
+      });
+    });
+
+    it('should assign PR author', () => {
+      expect(github.issues.addAssignees).toHaveBeenCalled();
+      expect(github.issues.addAssignees).toHaveBeenCalledWith({
+        issue_number: payloadData.payload.pull_request.number,
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        assignees: [payloadData.payload.pull_request.user.login],
+      });
+    });
+  });
+
+  describe('when pull request contains description and filled karma check', () => {
```

Comment:
This doesn't match the test.

Reply:
It does. The `bodyWithExplanationAndLint` checks the `linter/karma` check.

Reply:
Ah sorry, I meant also add in the description that what is missing - actually just that would be sufficient (in the variable names too).